packages/editor-ui/src/views/WorkflowsView.vue

```diff
@@ -243,13 +245,45 @@ const WorkflowsView = defineComponent({
 		sendFiltersTelemetry(source: string) {
 			(this.$refs.layout as IResourcesListLayoutInstance).sendFiltersTelemetry(source);
 		},
+		saveFiltersOnQueryString() {
+			const queryString = new URLSearchParams();
+
+			if (this.filters.search) {
+				queryString.append('search', this.filters.search);
+			}
+
+			if (this.filters.status !== undefined && this.filters.status !== '') {
+				queryString.append('status', this.filters.status);
+			}
+
+			if (this.filters.tags.length) {
+				queryString.append('tags', this.filters.tags.join(','));
+			}
+
+			void this.$router.push({
+				name: VIEWS.WORKFLOWS,
+				query: { ...Object.fromEntries(queryString) },
+			});
+		},
+		setFiltersFromQueryString() {
+			const { tags, status, search } = this.$route.query;
+
+			this.filters = {
+				...this.filters,
+				...(search && { search }),
+				...(status && { status: !!status }),
```

Comment:
status is a string no? `StatusFilter.ALL`  for example? and we should probably validate it too...

Reply:
StatusFilter.ALL is empty. With this condition it will not be included in the filter and if not included the default is empty!

<img width="366" alt="image" src="https://github.com/n8n-io/n8n/assets/16496553/eddb17e9-09a5-4dc8-bb44-f38d4015a04a">